packages/langium/src/generator/template-node.ts

```diff
@@ -0,0 +1,290 @@
+                    : (result, line, j) =>
+                        // handling of the remaining template segments
+                        j === 0 ? [line] : result.concat(NEWLINE, line) // except for the first line in the current segment prepend each line with NEWLINE
+                , [] // start with an empty array
+            ).filter(
+                e => !(typeof e === 'string' && e.length === 0)         // drop empty strings, they don't contribute anything but might confuse subsequent processing
+            ).concat(
+                // append the corresponding substitution after each segment (part),
+                //  note that 'substitutions[i]' will be undefined for the last segment
+                isGeneratorNode(substitutions[i])
+                    // if the substitution is a generator node, take it as it is
+                    ? substitutions[i] as GeneratorNode
+                    : substitutions[i] !== undefined
+                        // if the substitution is something else, convert it to a string and wrap it in a node;
+                        //  allows us below to distinghuish template strings from substitution (esp. empty) ones
+                        ? new CompositeGeneratorNode(String(substitutions[i]))
+                        : i < substitutions.length
+                            // if 'substitutions[i]' is undefined and we are treating a substitution "in the middle"
+                            //   we found a substition that is assumed to not contribute anything on purpose!
+                            ? UNDEFINED_SEGMENT  // add a corresponding marker, see below for details on the rational
+                            : []                 /* don't concat anything as we passed behind the last substitution, since 'i' enumerates the indices of 'staticParts',
+                                                     but 'substitutions' has one entry less and 'substitutions[staticParts.length -1 ]' will always be undefined */
+            )
+        );
+    });
+
+    // for templates like
+    //   expandToNode`
+    //    someText
+    //   `
+
+    const splitAndMergedLength = splitAndMerged.length;
+    const lastItem = splitAndMergedLength !== 0 ? splitAndMerged[splitAndMergedLength-1] : undefined;
+
+    if ((omitLastLine || trimLastLine) && typeof lastItem === 'string' && lastItem.trim().length === 0) {
+        if (omitFirstLine && splitAndMergedLength !== 1 && splitAndMerged[splitAndMergedLength-2] === NEWLINE) {
+            return splitAndMerged.slice(0, splitAndMergedLength-2);
+        } else {
+            return splitAndMerged.slice(0, splitAndMergedLength-1);
+        }
+    } else {
+        return splitAndMerged;
+    }
+}
+
+export interface ForEachOptions<T> {
+    prefix?: (element: T, index: number, isLast: boolean) => Generated|undefined;
+    suffix?: (element: T, index: number, isLast: boolean) => Generated|undefined;
+    appendNewLineIfNotEmpty?: true;
+}
+
+export function forEach<T>(
```

Comment:
This name is ambiguous in the global package namespace.

A possible alternative would be to wrap some generator-related functions in a `Generator` namespace in the main `index.ts`.